### ISSUE TITLE
[Feature Request] parameter to rate limit youtube api requests

### ISSUE THREAD

**Issue by bani6809** [yt-dlp/yt-dlp#106]
## Checklist

- [X ] I'm reporting a feature request
- [X ] I've verified that I'm running yt-dlp version **2021.02.19**
- [X ] I've searched the bugtracker for similar feature requests including closed ones

## Description

WRITE DESCRIPTION HERE

there needs to be a feature to rate limit youtube api requests. for channels with thousands of videos, youtube-dlp downloads hundreds or thousands of channel index pages at full speed, eventually resulting in the dreaded 429 blocking from youtube.

--sleep-interval is insufficient as this only pauses in between video downloads. it does not prevent youtube-dlp from blasting youtube with api calls at full speed for everything else.

the solution is actually described here, but it has not been implemented yet: https://github.com/blackjack4494/yt-dlc/issues/158#issuecomment-724825796

basically there needs to be a parameter to rate limit youtube api requests so we do not get 429 blocked.

**Comment by pukkandan** [comment]
I am aware of the issue (and the proposed solution). But correctly implementing this is a bit tricky

**Comment by pukkandan** [comment]
Related: https://github.com/ytdl-org/youtube-dl/pull/28270

**Comment by bani6809** [comment]
> I can confirm my PR fixes this issue for my massive archive, feel free to use it.

i confirm this fixes my issues as well. yt-dlp is now usable again, and 429 is now a thing of the past for me.

finally i can download from huge channels again.
